save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/basic/ManageUserRoleCard.kt

```diff
@@ -0,0 +1,276 @@
+/**
+ * Components for cards
+ */
+
+@file:Suppress("FILE_NAME_MATCH_CLASS", "MatchingDeclarationName")
+
+package org.cqfn.save.frontend.components.basic
+
+import org.cqfn.save.domain.Role
+import org.cqfn.save.frontend.externals.fontawesome.faTimesCircle
+import org.cqfn.save.frontend.externals.fontawesome.fontAwesomeIcon
+import org.cqfn.save.frontend.externals.lodash.debounce
+import org.cqfn.save.frontend.utils.*
+import org.cqfn.save.info.UserInfo
+import org.cqfn.save.permission.SetRoleRequest
+
+import csstype.None
+import org.w3c.dom.HTMLInputElement
+import org.w3c.dom.HTMLSelectElement
+import org.w3c.fetch.Headers
+import org.w3c.fetch.Response
+import react.*
+import react.dom.*
+import react.dom.onClick
+
+import kotlinx.html.*
+import kotlinx.html.js.onChangeFunction
+import kotlinx.html.js.onClickFunction
+import kotlinx.js.jso
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+
+/**
+ * [RProps] for card component
+ */
+external interface ManageUserRoleCardProps : Props {
+    /**
+     * Information about user who is seeing the view
+     */
+    var selfUserInfo: UserInfo
+
+    /**
+     * Full name of a group
+     */
+    var groupPath: String
+
+    /**
+     * Kind of a group that will be shown ("project" or "organization" for now)
+     */
+    var groupType: String
+}
+
+// private fun String.toRole() = when (this) {
+// Role.VIEWER.toString(), Role.VIEWER.formattedName -> Role.VIEWER
+// Role.SUPER_ADMIN.toString(), Role.SUPER_ADMIN.formattedName -> Role.SUPER_ADMIN
+// Role.OWNER.toString(), Role.OWNER.formattedName -> Role.OWNER
+// Role.ADMIN.toString(), Role.ADMIN.formattedName -> Role.ADMIN
+// Role.NONE.toString(), Role.NONE.formattedName -> Role.NONE
+// else -> throw IllegalStateException("Unknown role is passed: $this")
+// }
+
+private fun String.toRole() = Role.values().find {
+    this == it.formattedName || this == it.toString()
+} ?: throw IllegalStateException("Unknown role is passed: $this")
+
+/**
+ * A functional `RComponent` for a card that shows users from the group and their permissions.
+ *
+ * @param updateErrorMessage
+ * @param getUserGroups
+ * @return a functional component representing a role managing card
+ */
+@Suppress(
+    "LongMethod",
+    "EMPTY_BLOCK_STRUCTURE_ERROR",
+    "TOO_LONG_FUNCTION",
+    "MAGIC_NUMBER",
+    "ComplexMethod",
+)
+fun manageUserRoleCardComponent(
```

Comment:
Any screenshots of the new components in PR description?

Reply:
Ok, Uploaded to PR.